spec/custom_matchers.rb

```diff
@@ -0,0 +1,12 @@
+RSpec::Matchers.define :belong_to do |expected|
+  match { |actual| (actual.to_a.uniq - expected.to_a.uniq).empty? }
+end
+
+RSpec::Matchers.define :contain_from do |expected|
+  match do |actual|
+    actual = actual.to_a.map { |x| x.data.to_a }.flatten.uniq
+    expected.map!(&:values) unless expected.first.is_a? Array
+    expected = expected.flatten.uniq
+    (actual - expected).empty?
+  end
+end
```

Comment:
Basically, Redis doesn't necessarily insert keys in the given order. Due to this, some rows and columns might be jumbled, and there is no way to expect for an exact match while testing on RSpec. Rather, the DataFrame is tested to have the same (or a subset, if specific keys / pattern is given) data, in **ANY** order.